### PR TITLE
Refactoring and stuff

### PR DESCRIPTION
**Pull request by harvey-taylor-bluefruit**

### CODE REVIEW

File: ConsoleApplicationGoBoard/Console.cpp [hunk 1]
```diff
@@ -6,28 +6,28 @@ using namespace ::std;
...
-				break;
-			case stone::white:
-				cout << "0 ";
-				break;
-			}
-		}
-		cout << endl;
-	}
-	Console::OutputDeadStones(goBoard);
+   board currentBoard = goBoard.GoBoard();
```
Comment (archived-joshua-bierton-bluefruit) [inline ConsoleApplicationGoBoard/Console.cpp]: goBoard.GoBoard is still confusing

File: ConsoleApplicationGoBoard/Console.cpp [hunk 2]
```diff
@@ -38,57 +38,57 @@ const void Console::OutputDeadStones(GameState &goBoard)
...
+   uint8_t size = goBoard.GoBoardSize();
+   int8_t x = size;
+   int8_t y = size;
+   while (x >= 0 && x >= size)
+   {
+      x = GetInput("please input an x coordinate : ");
+   }
+   while (y >= 0 && y >= size)
+   {
+      y = GetInput("please input an y coordinate : ");
```
Comment (archived-joshua-bierton-bluefruit) [inline ConsoleApplicationGoBoard/Console.cpp]: should be "a y coordinate"

File: Go Client/GameState.cpp [hunk 3]
```diff
@@ -63,29 +63,29 @@ stone GameState::WhosMoveJustWent()
...
 uint16_t GameState::PrizonerStones(stone stone)
 {
-	switch (stone)
-	{
-	case stone::white:
+   switch (stone)
+   {
+   case stone::white:
       return m_deadWhiteStones;
       break;
```
Comment (archived-joshua-bierton-bluefruit) [inline Go Client/GameState.cpp]: the break after the return is unreachable code and may cause warnings on some compilers and that sort of thing

File: Go Client/GameState.cpp [hunk 4]
```diff
@@ -63,29 +63,29 @@ stone GameState::WhosMoveJustWent()
 
 uint16_t GameState::PrizonerStones(stone stone)
```
Comment (archived-joshua-bierton-bluefruit) [inline Go Client/GameState.cpp:65]: inconsistent naming here: could probably stick to either prisoner stones or dead stones instead of both?

Comment (archived-joshua-bierton-bluefruit) [inline Go Client/GameState.cpp:65]: also could be prefixed with Get to make it's purpose more obvious?

File: Go Client/Go Rules.cpp [hunk 5]
```diff
@@ -1,93 +1,93 @@
...
+   {
+      for (uint8_t y = 0; y < m_currentBoard.size(); ++y)
+      {
+         CheckRules(x, y, goBoard, stoneChecking);
+      }
+   }
 }
 
-void Rules::CheckRules(uint8_t x, uint8_t y, GameState &goBoard, stone StoneChecking)
+void Rules::CheckRules(uint8_t x, uint8_t y, GameState &goBoard, stone stoneChecking)
```
Comment (archived-joshua-bierton-bluefruit) [inline Go Client/Go Rules.cpp]: not very clear what the purpose of stoneChecking is from it's name. colourToCheck?
also the function isn't just checking the rules, it's also enforcing them, so maybe that should be renamed

File: ConsoleApplicationGoBoard/Console.cpp [hunk 6]
```diff
@@ -6,28 +6,28 @@ using namespace ::std;
 
 const void Console::PrintBoard(GameState &goBoard)
 {
```
Comment (laurent-rabatel-bluefruit) [inline ConsoleApplicationGoBoard/Console.cpp:8]: const GameState &goBoard?

File: ConsoleApplicationGoBoard/Console.cpp [hunk 7]
```diff
@@ -38,57 +38,57 @@ const void Console::OutputDeadStones(GameState &goBoard)
...
+   {
+      y = GetInput("please input an y coordinate : ");
+   }
 
-	return goBoard.PlayStone(x, y, goBoard.WhosMoveIsNext());
+   return goBoard.PlayStone(x, y, goBoard.WhosMoveIsNext());
 }
 
 uint8_t Console::GetInput(string message)
 {
```
Comment (laurent-rabatel-bluefruit) [inline ConsoleApplicationGoBoard/Console.cpp:65]: const string &message?

File: ConsoleApplicationGoBoard/Console.cpp [hunk 8]
```diff
@@ -38,57 +38,57 @@ const void Console::OutputDeadStones(GameState &goBoard)
...
+      }
+      catch (out_of_range)
+      {
+         cout << "Out of range!" << endl;
+         cout << message << endl;
+         succesfull = false;
+      }
+   }
+   return input;
 }
```
Comment (laurent-rabatel-bluefruit) [inline ConsoleApplicationGoBoard/Console.cpp:94]: so this returns 0 if not successful, is that OK?

File: Go Client/GameState.cpp [hunk 9]
```diff
@@ -36,19 +36,19 @@ board GameState::GoBoard()
 bool GameState::PlayStone(uint8_t X, uint8_t Y, stone stoneColour)
 {
    if (m_goBoard[Y][X] != stone::empty && stoneColour != stone::empty)
-		return false;
+      return false;
```
Comment (laurent-rabatel-bluefruit) [inline Go Client/GameState.cpp:39]: Can I suggest braces or empty line just after to isolate this statement?

File: Go Client/GameState.cpp [hunk 10]
```diff
@@ -63,29 +63,29 @@ stone GameState::WhosMoveJustWent()
...
       break;
    case stone::empty:
       return 0;
       break;
-	}
+   }
 }
 
 void GameState::IncrementDeadStones(stone stone)
 {
```
Comment (laurent-rabatel-bluefruit) [inline Go Client/GameState.cpp:78]: this looks a bit weird: variable name is exactly the same as type name

File: Go Client/GameState.h [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+   void IncrementDeadStones(stone stone);
+private:
+   uint16_t m_deadWhiteStones;
+   uint16_t m_deadBlackStones;
+   board CreateGoBoard();
+   const uint8_t m_goBoardSize;
+   board m_goBoard;
+   stone m_turn;
+   stone m_notTurn;
+};
```
Comment (laurent-rabatel-bluefruit) [inline Go Client/GameState.h:24]: What is the use of m_notTurn?

File: Go Client.Tests/GoRules_Tests.cpp [hunk 12]
```diff
@@ -4,98 +4,98 @@
...
+   testBoard.PlayStone(0, 1, stone::white);
+   testBoard.PlayStone(2, 2, stone::black);
+   testBoard.PlayStone(0, 2, stone::white);
+   testBoard.PlayStone(1, 1, stone::black);
+   testBoard.PlayStone(1, 2, stone::white);
+   testBoard.PlayStone(1, 3, stone::black);
+   testrules.DoRules(testBoard.WhosMoveIsNext(), testBoard);
+   testrules.DoRules(testBoard.WhosMoveJustWent(), testBoard);
+   EXPECT_EQ(expectedboard, testBoard.GoBoard());
 }
```
Comment (laurent-rabatel-bluefruit) [inline Go Client.Tests/GoRules_Tests.cpp:101]: :+1:

File: Go Client/Go Rules.cpp [hunk 13]
```diff
@@ -1,93 +1,93 @@
 #include "Go Rules.h"
```
Comment (laurent-rabatel-bluefruit) [inline Go Client/Go Rules.cpp:2]: I like to avoid spaces in file names, it can cause problems with some tools

File: Go Client/Go Rules.cpp [hunk 14]
```diff
@@ -1,93 +1,93 @@
...
+   goBoard.PlayStone(x, y, stone::empty);
+   m_currentBoard = goBoard.GoBoard();
+   if (x < m_currentBoard.size() - 1)
+      if (m_currentBoard[y][x + 1] == m_currentStone)
+         KillGroup(x + 1, y, goBoard);
+   if (y < m_currentBoard.size() - 1)
+      if (m_currentBoard[y + 1][x] == m_currentStone)
+         KillGroup(x, y + 1, goBoard);
+   if (x)
+      if (m_currentBoard[y][x - 1] == m_currentStone)
```
Comment (laurent-rabatel-bluefruit) [inline Go Client/Go Rules.cpp:47]: I think braces would help here :)

File: Go Client/Go Rules.cpp [hunk 15]
```diff
@@ -1,93 +1,93 @@
...
+   if (x < m_currentBoard.size() - 1)
+      if (m_currentBoard[y][x + 1] == m_currentStone)
+         NumberOflibertiesOfAGroup(x + 1, y);
+   if (y < m_currentBoard.size() - 1)
+      if (m_currentBoard[y + 1][x] == m_currentStone)
+         NumberOflibertiesOfAGroup(x, y + 1);
+   if (x)
+      if (m_currentBoard[y][x - 1] == m_currentStone)
+         NumberOflibertiesOfAGroup(x - 1, y);
+   if (y)
```
Comment (laurent-rabatel-bluefruit) [inline Go Client/Go Rules.cpp:77]: NumberOf_L_ibertiesOfAGroup?
I think the name of this method is a bit confusing - to me, it looks very similar to CheckLibertiesOfAStone, and if it doesn't return anything, maybe it should be called CheckLibertiesOfAGroup?

File: ConsoleApplicationGoBoard/Console.cpp (hunk 8 above)
Comment (harvey-taylor-bluefruit) [inline ConsoleApplicationGoBoard/Console.cpp:94]: untill successful = true then it dosent leave this code :) im confused at what you are trying to say :P

File: Go Client/Go Rules.cpp (hunk 15 above)
Comment (harvey-taylor-bluefruit) [inline Go Client/Go Rules.cpp:77]: it does return the number of liberties of a group but also increments m_leberties... ummm yeah CheckNumberOfLibertiesOfAGroup is better tho :)

File: Go Client/GameState.h (hunk 11 above)
Comment (harvey-taylor-bluefruit) [inline Go Client/GameState.h:24]: i think its because i was lazy and i was using it elsewhere :/ whoops :)

File: Go Client/GameState.cpp (hunk 10 above)
Comment (harvey-taylor-bluefruit) [inline Go Client/GameState.cpp:78]: i agree

File: ConsoleApplicationGoBoard/Console.cpp (hunk 8 above)
Comment (laurent-rabatel-bluefruit) [inline ConsoleApplicationGoBoard/Console.cpp:94]: I'm wondering whether 0 is a safe return value for an error... isn't it a valid input as well?

**Pull request merged**